src/components/PopupWithForm.js

```diff
@@ -18,17 +19,51 @@ export default class PopupWithForm extends Popup {
     return formValues;
   }
 
+  _checkFormValidity() {
+    const isFormValid = Array.from(this._inputList).every(
+      (input) => input.value.trim() !== ""
+    );
+    this._submitButton.disabled = !isFormValid;
+    if (isFormValid) {
+      this._submitButton.classList.remove("button_inactive");
+    } else {
+      this._submitButton.classList.add("button_inactive");
+    }
+  }
+
   _setEventListeners() {
-    this._popupForm.addEventListener("submit", (event) => {
+    this._popupForm.addEventListener("submit", async (event) => {
       event.preventDefault();
       const data = this._getInputValues();
-      this._handleFormSubmission(data);
-      this.reset();
+      this.renderLoading(true);
+
+      await this._handleFormSubmission(data);
+
+      this.close();
```

Comment:
* You haven't corrected my remark https://github.com/saralwraz/se_project_aroundtheus/pull/1#discussion_r1739596759

Reply:
```js

      this.close();
      this.renderLoading(false);
```

Please, delete them. You have everything in `index.js`